pcsx2-qt/Debugger/MemoryViewWidget.cpp

```diff
@@ -225,6 +225,26 @@ void MemoryViewTable::InsertIntoSelectedHexView(u8 value)
 		QtHost::RunOnUIThread([this] { parent->update(); });
 	});
 }
+
+void MemoryViewTable::InsertAtCurrentSelection(const QString& text) {
+	if (!m_cpu->isValidAddress(selectedAddress))
+		return;
+
+	// Convert the input to ASCII.
+	// If pasting into the hex view, also decode the input as hex bytes.
+	// This approach prevents one from pasting on a nibble boundary, but that is almost always
+	// user error, and we don't have an undo function in this view, so best to stay conservative.
+	QByteArray input = selectedText ? text.toLatin1() : QByteArray::fromHex(text.toLatin1());
```

Comment:
Any reason specifically to use latin1 here? We use UTF-8 everywhere across the code base, generally.

Reply:
The rationale was to prevent Unicode characters which aren't valid ASCII characters from being pasted into a context where Unicode is rarely, if ever used (in this case, the PS2's memory). But it also has the convenient side effect of letting us effectively `memcpy()` the pasted input into the CPU memory for both the "text" and "hex" use cases.

Reply:
The Qt document says that the "returned byte array is undefined if the string contains non-Latin1 characters", so that's a little concerning.

If you're only intending to allow ASCII, then it might be worth validating that no characters are beyond 127 before converting (plus, then could use the utf8 routine instead, for consistency).

Reply:
Actually, now that I think about it, it makes sense to just use UTF-8 and not try to clamp to ASCII (in addition to the reasons you mentioned):

- the `Copy Character` operation in the memory view actually copies UTF-8, not individual ASCII characters
- I only considered the case of pasting text from a hex editor, but the real use case for text paste would be pasting from something like a text document
- NTSC-J games might contain UTF-8 encoded Japanese text

I've pushed an update to the PR switching to UTF-8.